File: appendString.java
public class appendString {
    public int appendCharacters(String s, String t) {
        int index1 = 0, index2 = 0;
        while (index1 < s.length() && index2 < t.length()) {
            if (s.charAt(index1) == t.charAt(index2)) index2++;
            index1++;
        }
        return t.length() - index2;
    }

    public static void main(String[] args) {
        String s = "z", t = "abcde";
        appendString appendString = new appendString();
        System.out.println(appendString.appendCharacters(s, t));
    }
}

File: appendString.py
class Solution:
    def appendCharacters(self, s: str, t: str) -> int:
        index1, index2 = 0, 0
        while index1 < len(s) and index2 < len(t):
            if s[index1] == t[index2]:
                index2+=1
            index1+=1
        return len(t) - index2